app/models/forms/delete_logs_form.rb

```diff
@@ -0,0 +1,18 @@
+module Forms
+  class DeleteLogsForm
+    include ActiveModel::Model
+    include ActiveModel::Validations
+
+    attr_reader :logs, :log_type
+
+    def initialize(logs: nil, log_ids: nil, log_type: nil)
+      @logs = logs
+      @log_ids = log_ids
+      @log_type = log_type || (logs.first&.lettings? ? :lettings : :sales)
```

Comment:
I think this optimisation has introduced complexity. Did you consider two separate form objects and templates for now?